### DISCUSSION TITLE
Rationale behind cidr-notation on a single ip

### DISCUSSION THREAD

**Discussion by xeor** [netbox-community/netbox#10583] [General]
When creating a single ip in netbox (`/ipam/ip-addresses/...`), it clearly states that the address must be defined as `IPv4 or IPv6 address (with mask)`.

I have looked around to find the rationale behind this.

* An ip belongs to a subnet anyway, shouldn't this mask be calculated?
* It feels very redundant
* Netbox won't stop me creating ip's with mask outside of it's subnet.
* It is somewhat confusing that this is a requirement.

I got some theories tho
* It is technically valid to define a host with an ip and subnet that is outside what it really belongs to. Not sure if there is any reason tho.
* It makes the database and query design more optimised
* It makes the ipv6 way of defining an ip more in-line with the ipv4 way.

Please help me understand this little design-choice

**Comment by kkthxbye-code** [comment]
Not sure if you saw, it's not easy to find, but there's some rationale here:

https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-does-netbox-require-every-ip-to-have-a-mask

**Reply by xeor** [reply]
doh.. I didn't see that. Thanks!
